src/Ilios/CliBundle/Command/SendTeachingRemindersCommand.php

```diff
@@ -0,0 +1,251 @@
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ilios:messaging:send-teaching-reminders')
+            ->setDescription('Sends teaching reminders to educators.')
+            ->addOption(
+                'base_url',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'The base URL of your Ilios instance.'
+            )
+            ->addOption(
+                'sender',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Email address to send reminders from.'
+            )
+            ->addOption(
+                'days',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'How many days in advance of teaching events reminders should be sent.',
+                7
+            )
+            ->addOption(
+                'subject',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'The subject line of reminder emails.',
+                'Upcoming Teaching Session'
+            )
+            ->addOption(
+                'dry-run',
+                null,
+                InputOption::VALUE_NONE,
+                'Prints out notification instead of emailing it. Useful for testing/debugging purposes.'
+            );
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        // input validation
+        $errors = $this->validateInput($input);
+        if (! empty($errors)) {
+            foreach ($errors as $error) {
+                $output->writeln("<error>{$error}</error>");
+            }
+            return;
+        }
+
+        $daysInAdvance = $input->getOption('days');
+        $sender = $input->getOption('sender');
+        $baseUrl = rtrim($input->getOption('base_url'), '/');
+        $subject = $input->getOption('subject');
+        $isDryRun = $input->getOption('dry-run');
+
+        // get all applicable offerings.
+        $offerings = $this->offeringManager->getOfferingsForTeachingReminders($daysInAdvance);
+
+        if ($offerings->isEmpty()) {
+            $output->writeln('<info>No offerings with pending teaching reminders found.</info>');
+            return;
+        }
+
+        // mail out a reminder per instructor per offering.
+        $templateCache = [];
+        $iterator = $offerings->getIterator();
+        $i = 0;
+
+        /** @var OfferingInterface $offering */
+        foreach ($iterator as $offering) {
+            $school = $offering->getSession()->getCourse()->getSchool();
+            if (! array_key_exists($school->getId(), $templateCache)) {
+                $template = $this->getTemplatePath($school);
+                $templateCache[$school->getId()] = $template;
+            }
+            $template = $templateCache[$school->getId()];
+
+            $instructors = $this->getAllInstructorsForOffering($offering);
```

Comment:
The offering entity now has a getAllInstructors method:
https://github.com/ilios/ilios/blob/master/src/Ilios/CoreBundle/Entity/Offering.php#L439